analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -97,7 +97,7 @@ class SpdxDocumentFile(
     analyzerConfig: AnalyzerConfiguration,
     repoConfig: RepositoryConfiguration
 ) : PackageManager(managerName, analysisRoot, analyzerConfig, repoConfig) {
-    var packagesFromExternalDocumentReferences = mutableSetOf<SpdxPackage>()
+    var packagesFromExternalDocumentReferences = mutableMapOf<String, SpdxPackage>()
```

Comment:
Consider using a variable name that makes more clear from what to what this maps.